lib/screenplay/config/s3_fetch.ex

```diff
@@ -121,4 +124,28 @@ defmodule Screenplay.Config.S3Fetch do
         {:error, :config_not_written}
     end
   end
+
+  defp add_labels_to_config(config) do
+    with {:ok, paess_labels_content, _} <- do_get(:paess_labels),
+         {:ok, paess_labels_json} <- Jason.decode(paess_labels_content) do
+      Enum.reduce(
+        config,
+        [],
+        fn place, acc ->
+          new_screens =
+            Enum.reduce(
+              place["screens"],
+              [],
+              fn screen, acc ->
+                [Map.put(screen, "label", Map.get(paess_labels_json, screen["id"])) | acc]
+              end
+            )
+
+          [Map.put(place, "screens", new_screens) | acc]
+        end
+      )
+    else
+      _ -> config
+    end
+  end
```

Comment:
**suggestion:** The contents of this `with` clause are the same as above. Can we pull that into a shared function?

Reply:
Yeah good idea. I feel like the best place for this is in `Screenplay.Config.Fetch` but wasn't sure if implementation and callbacks should live in the same module. What do you think?

Reply:
I think that's totally fine!